packages/descendants/src/Descendants/useInitDescendants.tsx

```diff
@@ -41,9 +51,26 @@ export const useInitDescendants = <
     [getDescendants, setDescendants],
   );
 
+  // Create a new Provider for the given context
+  const Provider = useMemo(() => {
+    // eslint-disable-next-line react/display-name
+    return ({ children }: PropsWithChildren<{}>) => {
+      return (
+        <DescendantsProvider
+          context={context}
+          descendants={getDescendants()}
```

Comment:
should this be using the `descendants` instead of `getDescendants()`?

Reply:
I thought so too, but no. We can't use the descendants object here since it would require us to add `descendants` as a dependency of this `useMemo`, resulting in a _different_ Provider component every time the descendants change. We need this to remain stable across renders. 
(Also, if we just omit `descendants` from the deps array, then it will refer to a stale vlaue)

Reply:
See the [latest changes](https://github.com/mongodb/leafygreen-ui/blob/cd49f1ddb3abf0c818ac53625bb83a9ec4623f54/packages/descendants/src/Highlight/useHighlight.tsx#L104-L113) to `useHighlight` where I have to do something similar to ensure the Provider stays stable across changes to `highlight`

Reply:
separately, I wonder if we could use a different naming between `descendants` and `getDescendants` to make it more explicit when to use which. Maybe something like `getStableDescendants`? Or maybe it even makes sense to only surface the latter; I'm not sure when `descendants` would make more sense over ` getDescendants`

Reply:
I'll ad guidance to prefer using `getDescendants`, since in most cases that's what would guarantee you the accurate value. In some cases you can get away with just referencing `descendants`, but I'm not sure if that would actually gain you anything